src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -76,6 +76,23 @@ static const struct cursor_alternatives cursors[] = {
 
 namespace mf = mir::frontend;
 
+namespace
+{
+template<typename T>
+auto data_buffer_to_debug_string(T* data, size_t elements) -> std::string
+{
+    std::string result = "[";
```

Comment:
I know this is only debug code, but this would be better as a `std::stringstream` rather than successive allocation / strcat.